indico/modules/events/sessions/controllers/management/sessions.py

```diff
@@ -0,0 +1,114 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2015 European Organization for Nuclear Research (CERN).
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+from flask import request
+
+from indico.core.db.sqlalchemy.colors import ColorTuple
+from indico.modules.events.sessions.controllers.management import RHManageSessionsBase, RHManageSessionBase
+from indico.modules.events.sessions.forms import CreateSessionForm
+from indico.modules.events.sessions.models.sessions import Session
+from indico.modules.events.sessions.operations import create_session, update_session, delete_sessions
+from indico.modules.events.sessions.util import get_colors, get_active_sessions
+from indico.modules.events.sessions.views import WPManageSessions
+from indico.web.flask.templating import get_template_module
+from indico.web.forms.base import FormDefaults
+from indico.web.util import jsonify_data, jsonify_template
+from MaKaC.webinterface.rh.base import RH
+
+
+def _render_sessions_list(event):
+    tpl = get_template_module('events/sessions/management/_sessions_list.html')
+    return tpl.render_sessions_list(event, get_active_sessions(event), get_colors())
+
+
+class RHSessionsList(RHManageSessionsBase):
+    """Display list of all sessions within the event"""
+
+    def _process(self):
+        return WPManageSessions.render_template('management/sessions_list.html', self.event_new.as_legacy,
+                                                event=self.event_new, sessions=get_active_sessions(self.event_new),
+                                                default_colors=get_colors())
+
+
+class RHAddSession(RHManageSessionsBase):
+    """Add a session to the event"""
+
+    def _process(self):
+        form = CreateSessionForm()
```

Comment:
Here you might want to specify a default for the colors (random color from the list, but preferably a unused one).

```
used_colors = {s.colors for s in self.event_new.sessions.filter_by(is_deleted=False)}
unused_colors = set(get_colors()) - used_colors
colors = random.choice(tuple(unused_colors) or get_colors())
form = CreateSessionForm(obj=FormDefaults(colors=colors))
```